fix: [dns] fix hash function for network_url in ManagedZone and Policy resources

paulojmdias opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->

When someone is trying to attach a private DNS zone to multiple networks in different projects, if the network name is the same across some of those projects, the resource will only attach only one of that networks with the same name, even if they are on different projects.

Example:
- Project `project1` with networks `network1` and `network2` to be attached
- Project `project2` with networks `network2` and `network3` to be attached

The Terraform resource will only try to attach `network1` from `project1`, `network2` and `network3` from `project3` thus ignoring `network2` from `project1`

Updating the way how the resource is handling the hash will fix this issue.

```hcl
resource "google_dns_managed_zone" "x" {
  dns_name      = "dummy.com."
  force_destroy = true
  name          = "dummy.com"
  project       = "dummy-project"
  visibility    = "private"

  cloud_logging_config {
    enable_logging = false
  }

  private_visibility_config {
    networks {
      network_url = "https://www.googleapis.com/compute/v1/projects/project1/global/networks/network1"
    }
    networks {
      network_url = "https://www.googleapis.com/compute/v1/projects/project1/global/networks/network2"
    }
    networks {
      network_url = "https://www.googleapis.com/compute/v1/projects/project2/global/networks/network2"
    }
    networks {
      network_url = "https://www.googleapis.com/compute/v1/projects/project2/global/networks/network3"
    }
  }
}
```



<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
dns: fix hash function for network_url in ManagedZone and Policy resources
```

modular-magician commented:
Hello! I am a robot. It looks like you are a: Community Contributor ~Googler~ ~Core Contributor~. Tests will require approval to run.

@melinath, a repository maintainer, has been assigned to review your changes. If you have not received review feedback within 2 business days, please leave a comment on this PR asking them to take a look.

You can help make sure that review is quick by [doing a self-review](https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/) and by [running impacted tests locally](https://googlecloudplatform.github.io/magic-modules/get-started/run-provider-tests/).

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 2 files changed, 4 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 2 files changed, 4 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2982` 
Passed tests `2683` 
Skipped tests: `295` 
Affected tests: `4` 

#### Action taken 
<details> <summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSPolicy_dnsPolicyBasicExample|TestAccDNSManagedZone_dnsManagedZonePrivateExample|TestAccDNSManagedZone_privateUpdate|TestAccHealthcareDatasetIamPolicy </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c03990f7-33db-4ee3-a0e8-6a1ffff14ef4/recording/TestAccDNSPolicy_dnsPolicyBasicExample.log)]
`TestAccDNSManagedZone_dnsManagedZonePrivateExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c03990f7-33db-4ee3-a0e8-6a1ffff14ef4/recording/TestAccDNSManagedZone_dnsManagedZonePrivateExample.log)]
`TestAccDNSManagedZone_privateUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c03990f7-33db-4ee3-a0e8-6a1ffff14ef4/recording/TestAccDNSManagedZone_privateUpdate.log)]
`TestAccHealthcareDatasetIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c03990f7-33db-4ee3-a0e8-6a1ffff14ef4/recording/TestAccHealthcareDatasetIamPolicy.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c03990f7-33db-4ee3-a0e8-6a1ffff14ef4/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c03990f7-33db-4ee3-a0e8-6a1ffff14ef4/recording) for each test

melinath reviewed (changes requested):
This makes sense and looks like it should work, but it would be great to also extend some existing [test cases](https://googlecloudplatform.github.io/magic-modules/develop/test/) to exercise this for both impacted resources.

paulojmdias commented:
Thanks for the feedback @melinath.

Proper tests were added.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 6 files changed, 266 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 6 files changed, 266 insertions(+), 4 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-8716-old..auto-pr-8716) ( 8 files changed, 294 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2989` 
Passed tests `2688` 
Skipped tests: `296` 
Affected tests: `5` 

#### Action taken 
<details> <summary>Found 5 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample|TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample|TestAccDataprocClusterIamPolicy|TestAccDNSPolicy_dnsPolicyMultiprojectExample|TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/recording/TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample.log)]
`TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/recording/TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample.log)]
`TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/recording/TestAccDataprocClusterIamPolicy.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{red}{\textsf{Tests failed when rerunning REPLAYING mode:}}$ 
 `TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/build-log/replaying_build_after_recording/TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/replaying_after_recording/TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample.log)] 

Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.

Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/build-log/recording_build/TestAccDNSPolicy_dnsPolicyMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)]
`TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/766255d8-6a89-499a-876d-d46a647c9248/recording) for each test

melinath commented on mmv1/products/dns/ManagedZone.yaml:
```diff
@@ -69,6 +69,16 @@ examples:
+      network_1_name: 'network-1'
+      network_2_name: 'network-2'
+      project_1_name: 'project-1'
+      project_2_name: 'project-2'
```
```suggestion
      project_2_name: 'project-2'
    test_env_vars:
      org_id: :ORG_ID
      billing_account: :BILLING_ACCT
```

melinath commented on mmv1/products/dns/Policy.yaml:
```diff
@@ -32,6 +32,16 @@ examples:
+      network_1_name: 'network-1'
+      network_2_name: 'network-2'
+      project_1_name: 'project-1'
+      project_2_name: 'project-2'
```
```suggestion
      project_2_name: 'project-2'
    test_env_vars:
      org_id: :ORG_ID
      billing_account: :BILLING_ACCT
```

melinath reviewed (changes requested):
it looks like you need to define test_env_vars

paulojmdias commented:
Many thanks @melinath. Added.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 188 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 188 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2991` 
Passed tests `2693` 
Skipped tests: `296` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample|TestAccDNSPolicy_dnsPolicyMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/29c3e6de-e59a-4f9e-adb4-d8832eb31068/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/29c3e6de-e59a-4f9e-adb4-d8832eb31068/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)]
`TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/29c3e6de-e59a-4f9e-adb4-d8832eb31068/build-log/recording_build/TestAccDNSPolicy_dnsPolicyMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/29c3e6de-e59a-4f9e-adb4-d8832eb31068/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/29c3e6de-e59a-4f9e-adb4-d8832eb31068/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/29c3e6de-e59a-4f9e-adb4-d8832eb31068/recording) for each test

melinath commented on mmv1/templates/terraform/examples/dns_managed_zone_private_multiproject.tf.erb:
```diff
@@ -0,0 +1,63 @@
+  }
+}
+
+resource "google_project" "project-1" {
```
Error: 

```
Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 38, in resource "google_compute_network" "network-1_project-1":
          38:   project                 = google_project.project_1.project_id
        
        A managed resource "google_project" "project_1" has not been declared in the
        root module.
        
```

All of these need to use underscores to match the references to them.

```suggestion
resource "google_project" "project_1" {
```

paulojmdias replied:
🤦‍♂️ Thanks @melinath. Very nice catch!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 188 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 188 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2996` 
Passed tests `2698` 
Skipped tests: `296` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSPolicy_dnsPolicyMultiprojectExample|TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c757a8cc-0bf9-47de-9ca8-e933ec7ba769/build-log/recording_build/TestAccDNSPolicy_dnsPolicyMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c757a8cc-0bf9-47de-9ca8-e933ec7ba769/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)]
`TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c757a8cc-0bf9-47de-9ca8-e933ec7ba769/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c757a8cc-0bf9-47de-9ca8-e933ec7ba769/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c757a8cc-0bf9-47de-9ca8-e933ec7ba769/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c757a8cc-0bf9-47de-9ca8-e933ec7ba769/recording) for each test

paulojmdias commented:
@melinath is there any way to external users check the error logs from tests ? I try but I always have 403.

melinath reviewed (changes requested):
Unfortunately no - the logs may contain sensitive information. Sorry for the inconvenience. However, you should be able to run tests locally to reproduce the failures: https://googlecloudplatform.github.io/magic-modules/develop/run-tests/ Looks like they are currently failing due to the Compute API not being enabled on the projects, which I think can be fixed with the google_project_service resource

paulojmdias commented:
> Unfortunately no - the logs may contain sensitive information. Sorry for the inconvenience. However, you should be able to run tests locally to reproduce the failures: https://googlecloudplatform.github.io/magic-modules/develop/run-tests/ Looks like they are currently failing due to the Compute API not being enabled on the projects, which I think can be fixed with the google_project_service resource

Thank you @melinath. I understand that.

Already added the missing `google_project_service` resources.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 260 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 260 insertions(+), 4 deletions(-))

paulojmdias commented:
@melinath can you please share some insights from error log for me to fix it? I have some issues reproducing the tests locally.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 274 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 5 files changed, 279 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2998` 
Passed tests `2700` 
Skipped tests: `296` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSPolicy_dnsPolicyMultiprojectExample|TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/b800ee17-387d-4f3d-8193-678228fc0d28/build-log/recording_build/TestAccDNSPolicy_dnsPolicyMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/b800ee17-387d-4f3d-8193-678228fc0d28/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)]
`TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/b800ee17-387d-4f3d-8193-678228fc0d28/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/b800ee17-387d-4f3d-8193-678228fc0d28/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/b800ee17-387d-4f3d-8193-678228fc0d28/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/b800ee17-387d-4f3d-8193-678228fc0d28/recording) for each test

paulojmdias commented:
@melinath could you please share some information of the log output ? I can't reproduce the tests locally because I don't have permission in any organization to create a project.

melinath reviewed (changes requested):
Error is in `TestAccDNSPolicy_dnsPolicyMultiprojectExample`:

```
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 84, in resource "google_project_service" "compute_project-1":
          84:   project = google_project.project_1.project_id
        
        A managed resource "google_project" "project_1" has not been declared in the
        root module.
        
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 89, in resource "google_project_service" "compute_project-2":
          89:   project = google_project.project_2.project_id
        
        A managed resource "google_project" "project_2" has not been declared in the
        root module.
        
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 94, in resource "google_project_service" "dns_project-1":
          94:   project = google_project.project_1.project_id
        
        A managed resource "google_project" "project_1" has not been declared in the
        root module.
        
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 99, in resource "google_project_service" "dns_project-2":
          99:   project = google_project.project_2.project_id
        
        A managed resource "google_project" "project_2" has not been declared in the
        root module.
```

It looks like the same issue that you had previously fixed of the references needing to match the resource declarations: https://github.com/GoogleCloudPlatform/magic-modules/pull/8716/files#r1306209654 Not sure what happened.

melinath commented:
it looks like the fix was inadvertently reverted in https://github.com/GoogleCloudPlatform/magic-modules/commit/3a367fafd767b7d4b539408506bc3fdae3b5c779

FWIW I'd recommend standardizing on either `_` or `-` in terraform resource IDs rather than mixing them (with a preference for `_`) - makes it easier to read and harder to run into this kind of reference error.

paulojmdias commented:
> it looks like the fix was inadvertently reverted in [3a367fa](https://github.com/GoogleCloudPlatform/magic-modules/commit/3a367fafd767b7d4b539408506bc3fdae3b5c779)
> 
> FWIW I'd recommend standardizing on either `_` or `-` in terraform resource IDs rather than mixing them (with a preference for `_`) - makes it easier to read and harder to run into this kind of reference error.

Weird, that was unexpected.

Anyway, I have already updated everything to use only `_` on the resource names.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 274 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 274 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `3008` 
Passed tests `2710` 
Skipped tests: `296` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample|TestAccDNSPolicy_dnsPolicyMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c8253213-1839-4a50-b56e-c630c02f8115/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c8253213-1839-4a50-b56e-c630c02f8115/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)]
`TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c8253213-1839-4a50-b56e-c630c02f8115/build-log/recording_build/TestAccDNSPolicy_dnsPolicyMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c8253213-1839-4a50-b56e-c630c02f8115/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c8253213-1839-4a50-b56e-c630c02f8115/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/c8253213-1839-4a50-b56e-c630c02f8115/recording) for each test

paulojmdias commented:
Hi @melinath, sorry but I think I need your help again to check the error logs. I don't see a reason for the tests to fail. 🥲

melinath commented on mmv1/templates/terraform/examples/dns_managed_zone_private_multiproject.tf.erb:
```diff
@@ -0,0 +1,106 @@
+  name            = "<%= ctx[:vars]['project_1_name'] %>"
+  project_id      = "<%= ctx[:vars]['project_1_name'] %>"
+  org_id          = "<%= ctx[:test_env_vars]['org_id'] %>"
+  billing_account = "<%= ctx[:test_env_vars]['billing_acct'] %>"
```
Error is:

```
Final error: failed to enable services: failed on request preconditions: googleapi: Error 400: Billing account for project '151608109921' is not found. Billing must be enabled for activation of service(s) 'compute.googleapis.com,compute.googleapis.com,compute.googleapis.com' to proceed.
```

This is because the billing account is currently being set to an empty string: https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716#diff-cb1c089ebc19fab4c219531c321c3fb56cf78a08e66785dacc89bb1490558deaR329

The key here (and in the other places the billing account is referenced) needs to match the key from the yaml file.

```suggestion
  billing_account = "<%= ctx[:test_env_vars]['billing_account'] %>"
```

paulojmdias replied:
Thanks @melinath. Fixed. 👍

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 274 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 274 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `3009` 
Passed tests `2711` 
Skipped tests: `296` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample|TestAccDNSPolicy_dnsPolicyMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{red}{\textsf{Tests failed when rerunning REPLAYING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/build-log/replaying_build_after_recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/replaying_after_recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)] 

Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.

Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/49c5fa7b-0102-4176-96c5-b72f9b5e4c4e/recording) for each test

paulojmdias commented:
Sorry @melinath, but could you please share the logs with me again ?

melinath commented on mmv1/templates/terraform/examples/dns_managed_zone_private_multiproject.tf.erb:
```diff
@@ -0,0 +1,106 @@
+
+resource "google_project_service" "dns_project_1" {
+  project = google_project.project_1.project_id
+  service = "dns.googleapis.com"
```
```suggestion
  service = "dns.googleapis.com"
  depends_on              = [ 
    google_project_service.compute_project_1,
    google_project_service.dns_project_1,
  ]
```

melinath commented on mmv1/templates/terraform/examples/dns_managed_zone_private_multiproject.tf.erb:
```diff
@@ -0,0 +1,106 @@
+
+resource "google_project_service" "dns_project_2" {
+  project = google_project.project_2.project_id
+  service = "dns.googleapis.com"
```
```suggestion
  service = "dns.googleapis.com"
  depends_on              = [ 
    google_project_service.compute_project_1,
    google_project_service.dns_project_1,
  ]
```

melinath reviewed (changes requested):
It looks like this is failing because Terraform's order of applying `google_project_service` is non-deterministic. I think that most tests don't run into this problem because they are only enabling `google_project_service` on a single project. Possibly making sure that project 2 services are always enabled after project one will fix the problem

paulojmdias commented:
@melinath dependencies added.

melinath commented on mmv1/templates/terraform/examples/dns_managed_zone_private_multiproject.tf.erb:
```diff
@@ -0,0 +1,124 @@
+  depends_on              = [
+    google_project.project_1,
+    google_project_service.compute_project_1,
+    google_project_service.dns_project_1,
```
I think this won't work since it would give the resource a dependency on itself.

I'm pretty sure the dependency graph you'll need here & in the other example is:

- google_project_service.compute_project_2 _depends_on_ google_project_service.dns_project_1 and google_project_service.compute_project_1
- google_project_service.dns_project_2 _depends_on_ google_project_service.dns_project_1 and google_project_service.compute_project_1

(It looks like there are other tests in the providers already that have multiple `google_project_service` resources in a single project, so I think it would be okay to not have the `google_project_service.*_project_1` resources depend on both projects - but it also probably doesn't hurt.)

paulojmdias replied:
Yeah, I agree with you and I have fixed that.

By the way, the dependency on itself was your suggestion [there](https://github.com/GoogleCloudPlatform/magic-modules/pull/8716#discussion_r1310652197) and I think by mistake 😄 
Anyway, I have tested an _itself_ dependency in a data resource and it worked. I have removed the dependency since it also for me makes no sense.

melinath replied:
whoops, sorry for the confusion 😂  Thanks!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 306 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 306 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `3010` 
Passed tests `2711` 
Skipped tests: `296` 
Affected tests: `3` 

#### Action taken 
<details> <summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample|TestAccDNSPolicy_dnsPolicyMultiprojectExample|TestAccDataSourceGoogleServiceAccountJwt </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)]
`TestAccDataSourceGoogleServiceAccountJwt`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/recording/TestAccDataSourceGoogleServiceAccountJwt.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{red}{\textsf{Tests failed when rerunning REPLAYING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/build-log/replaying_build_after_recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/replaying_after_recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)] 

Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.

Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/build-log/recording_build/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/741d0087-29d1-497f-83b8-a28790c72313/recording) for each test

paulojmdias commented:
@melinath failed again. 😐 

Could you please share again the log output? Thanks

melinath commented:
TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample : Error: Error creating ManagedZone: googleapi: Error 400: Invalid value for 'entity.managedZone.dnsName': 'tf-test-private-multiproject.example.com.byp2cxtphv'

TestAccDNSPolicy_dnsPolicyMultiprojectExample: still failing for non-determinism. It could be worth making the dns enablement for each project depend_on the compute enablement. If that doesn't fix the issue then there's something else going on that I'll need to dig into.

Thank you for sticking with this! :-)

melinath reviewed (changes requested):
TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample:

```
Error: Error creating ManagedZone: googleapi: Error 400: Invalid value for 'entity.managedZone.dnsName': 'tf-test-private-multiproject.example.com.byp2cxtphv'
```

Does it need to use `_` for the zone_dns_name possibly?

TestAccDNSPolicy_dnsPolicyMultiprojectExample: still failing for non-determinism. It could be worth making the dns enablement for each project depend_on the compute enablement. If that doesn't fix the issue then there's something else going on that I'll need to dig into.

Thank you for sticking with this! :-)

paulojmdias commented:
> TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample:
> 
> ```
> Error: Error creating ManagedZone: googleapi: Error 400: Invalid value for 'entity.managedZone.dnsName': 'tf-test-private-multiproject.example.com.byp2cxtphv'
> ```
> 
> Does it need to use `_` for the zone_dns_name value possibly?
> 
> TestAccDNSPolicy_dnsPolicyMultiprojectExample: still failing for non-determinism. It could be worth making the dns enablement for each project depend_on the compute enablement. If that doesn't fix the issue then there's something else going on that I'll need to dig into.
> 
> Thank you for sticking with this! :-)

I have reverted the `dns_name` to a variable and used a fixed value of `multiproject.private.example.com.` to avoid those errors.

Also added the new dependency in DNS API enablement.

No worries, I just want to have the best option merged :)

melinath commented:
> I have reverted the dns_name to a variable and used a fixed value of multiproject.private.example.com. to avoid those errors.

That should be fine as long as the dns_name doesn't need to be unique across GCP or anything :-)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 308 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 308 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `3010` 
Passed tests `2712` 
Skipped tests: `296` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDNSPolicy_dnsPolicyMultiprojectExample|TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/42933aaf-2389-4880-93db-6abfe2aa298a/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)]
`TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/42933aaf-2389-4880-93db-6abfe2aa298a/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{red}{\textsf{Tests failed when rerunning REPLAYING mode:}}$ 
 `TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/42933aaf-2389-4880-93db-6abfe2aa298a/build-log/replaying_build_after_recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/42933aaf-2389-4880-93db-6abfe2aa298a/replaying_after_recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)] 

Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.

Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.

---
$\textcolor{red}{\textsf{Errors occurred during RECORDING mode. Please fix them to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/42933aaf-2389-4880-93db-6abfe2aa298a/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/42933aaf-2389-4880-93db-6abfe2aa298a/recording) for each test

paulojmdias commented:
@melinath failed again. Could you please check the logs and share with me ?

melinath commented:
It's the same error :-/

```
Error: Request `Enable Project Service "compute.googleapis.com" for project "tf-test-project-2nvphhjeek0"` returned error: Batch request and retried single request "Enable Project Service \"compute.googleapis.com\" for project \"tf-test-project-2nvphhjeek0\"" both failed. Final error: failed to enable services: failed on request preconditions: Post "https://serviceusage.googleapis.com/v1/projects/tf-test-project-2nvphhjeek0/services/compute.googleapis.com:enable?alt=json&prettyPrint=false": Requested interaction not found
```

I'll have to dig into it more.

melinath commented:
it's interesting though that `TestAccDNSPolicy_dnsPolicyMultiprojectExample` is now working as expected...

paulojmdias commented:
> it's interesting though that `TestAccDNSPolicy_dnsPolicyMultiprojectExample` is now working as expected...

Can't be some random errors ? Don't know if just retrying we will have the same behavior

melinath commented:
worth a shot /gcbrun

paulojmdias commented:
Don't know also if is related
- https://github.com/hashicorp/terraform-provider-google/issues/14158
- https://github.com/hashicorp/terraform-provider-google/issues/14160

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 308 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 308 insertions(+), 4 deletions(-))

melinath commented on mmv1/templates/terraform/examples/dns_policy_multiproject.tf.erb:
```diff
@@ -0,0 +1,117 @@
+resource "google_project_service" "dns_project_2" {
+  project = google_project.project_2.project_id
+  service = "dns.googleapis.com"
+  depends_on              = [
```
```suggestion
  depends_on              = [
    google_project_service.compute_project_2,
  ]
```

melinath commented on mmv1/templates/terraform/examples/dns_policy_multiproject.tf.erb:
```diff
@@ -0,0 +1,117 @@
+resource "google_project_service" "compute_project_2" {
+  project = google_project.project_2.project_id
+  service = "compute.googleapis.com"
+  depends_on              = [
```
```suggestion
  depends_on              = [
    google_project_service.dns_project_1
  ]
```

melinath commented on mmv1/templates/terraform/examples/dns_policy_multiproject.tf.erb:
```diff
@@ -0,0 +1,117 @@
+resource "google_project_service" "dns_project_1" {
+  project = google_project.project_1.project_id
+  service = "dns.googleapis.com"
+  depends_on              = [
```
```suggestion
  depends_on              = [
    google_project_service.compute_project_1
  ]
```

melinath reviewed (changes requested):
Sorry, looks like I wasn't clear about the dependencies and missed that they weren't updated properly.

The important thing is that the `google_project_service` resources execute in a deterministic order relative to each other. Specifically, changing the depends_on as suggested will result in:

1. google_project_service.compute_project_1
2. google_project_service.dns_project_1
3. google_project_service.compute_project_2
4. google_project_service.dns_project_2

Which should pass tests. You should make the same changes in the other example file.

paulojmdias commented:
> Sorry, looks like I wasn't clear about the dependencies and missed that they weren't updated properly.
> 
> The important thing is that the `google_project_service` resources execute in a deterministic order relative to each other. Specifically, changing the depends_on as suggested will result in:
> 
> 1. google_project_service.compute_project_1
> 2. google_project_service.dns_project_1
> 3. google_project_service.compute_project_2
> 4. google_project_service.dns_project_2
> 
> Which should pass tests. You'll need to delete the extra depends_on lines & make the same changes in the other example file.

Make sense @melinath.

I have added the required changes for that.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 298 insertions(+), 4 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8716-old..auto-pr-8716) ( 4 files changed, 298 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `3010` 
Passed tests `2711` 
Skipped tests: `296` 
Affected tests: `3` 

#### Action taken 
<details> <summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataprocJobIamPolicy|TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample|TestAccDNSPolicy_dnsPolicyMultiprojectExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocJobIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/e94c0506-52bd-4195-aca3-9dd5bfbd329a/recording/TestAccDataprocJobIamPolicy.log)]
`TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/e94c0506-52bd-4195-aca3-9dd5bfbd329a/recording/TestAccDNSManagedZone_dnsManagedZonePrivateMultiprojectExample.log)]
`TestAccDNSPolicy_dnsPolicyMultiprojectExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/e94c0506-52bd-4195-aca3-9dd5bfbd329a/recording/TestAccDNSPolicy_dnsPolicyMultiprojectExample.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/e94c0506-52bd-4195-aca3-9dd5bfbd329a/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8716/artifacts/e94c0506-52bd-4195-aca3-9dd5bfbd329a/recording) for each test

paulojmdias commented:
Finally everything is green @melinath 😎

paulojmdias commented:
@melinath can you review it please ? 🙏

melinath reviewed (approved):
LGTM - thanks for sticking with this!
